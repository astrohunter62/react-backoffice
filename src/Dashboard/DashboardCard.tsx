import React from "react";
import classNames from "classnames";
import {
  Card,
  CardContent,
  Typography,
  withStyles,
  Avatar,
} from "@material-ui/core";
import DisabledIcon from "@material-ui/icons/Lock";

const styles = (theme: any) => ({
  root: {
    cursor: "pointer",
  },
  content: {
    paddingBottom: `${theme.spacing(2)}px !important`,
  },
  avatar: {
    float: "left",
    marginRight: theme.spacing(2),
    backgroundColor: theme.palette.secondary.dark,
  },
  title: {
    marginBottom: 0,
  },
  disabled: {
    opacity: 0.75,
    pointerEvents: "none",
    filter: "grayscale(30%)",
  },
  icon: {
    float: "right",
    opacity: 0.5,
  },
});

type DashboardCardProps = {
  title: string;
  description?: string;
  handleClick: (...args: any[]) => any;
  icon?: (...args: any[]) => any;
  isDisabled?: boolean;
  classes: {
    [key: string]: string;
  };
};

const DashboardCard: React.SFC<DashboardCardProps> = ({
  title,
  description,
  handleClick,
  icon: Icon,
  isDisabled,
  classes,
}) => {
  const rootClasses = classNames({
    [classes.root]: true,
    [classes.disabled]: isDisabled,
  });

  return (
    <Card onClick={handleClick} className={rootClasses}>
      <CardContent className={classes.content}>
        {isDisabled ? <DisabledIcon className={classes.icon} /> : null}

        {Icon ? (
          <Avatar className={classes.avatar}>
            <Icon />
          </Avatar>
        ) : null}

        <Typography variant="h5" className={classes.title} component="h2">
          {title}
        </Typography>

        {description ? (
          <Typography variant="body2">{description}</Typography>
        ) : null}
      </CardContent>
    </Card>
  );
};
DashboardCard.defaultProps = {
  isDisabled: false,
};

export default withStyles(styles as any)(DashboardCard);
