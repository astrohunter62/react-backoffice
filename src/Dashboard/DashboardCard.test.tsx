import React from "react";
import { shallow } from "enzyme";

import DashboardCard from "./DashboardCard";

it("renders correctly", () => {
  const tree = shallow(
    <DashboardCard title="Title" description="desc" handleClick={() => {}} />,
  );

  expect(tree).toMatchSnapshot();
});

it("renders disabled", () => {
  const tree = shallow(
    <DashboardCard
      title="Title"
      description="desc"
      handleClick={() => {}}
      isDisabled
    />,
  );

  expect(tree).toMatchSnapshot();
});
