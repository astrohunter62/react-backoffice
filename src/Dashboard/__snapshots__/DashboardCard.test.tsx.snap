// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DashboardCard
  classes={
    Object {
      "avatar": "DashboardCard-avatar-3",
      "content": "DashboardCard-content-2",
      "disabled": "DashboardCard-disabled-5",
      "icon": "DashboardCard-icon-6",
      "root": "DashboardCard-root-1",
      "title": "DashboardCard-title-4",
    }
  }
  description="desc"
  handleClick={[Function]}
  isDisabled={false}
  title="Title"
/>
`;

exports[`renders disabled 1`] = `
<DashboardCard
  classes={
    Object {
      "avatar": "DashboardCard-avatar-3",
      "content": "DashboardCard-content-2",
      "disabled": "DashboardCard-disabled-5",
      "icon": "DashboardCard-icon-6",
      "root": "DashboardCard-root-1",
      "title": "DashboardCard-title-4",
    }
  }
  description="desc"
  handleClick={[Function]}
  isDisabled={true}
  title="Title"
/>
`;
